### DISCUSSION TITLE
Mutual tls client certificate authentication

### DISCUSSION THREAD

**Discussion by ippocratis** [Radiokot/photoprism-android-client#17] [Ideas]
I am reverse proxying photoprism with caddy and i have configured it to use mutual tls with selected services it proxies adding the corenspodimg directive to them

I have installed the p12 client cert to the amdroid store

With the PWA it was easy
since it opens in a browser and the browser picks the client cert from the device cert store 

Mutual tls will add an extra security layer to the app

Example app using it
https://github.com/bitfireAT/davx5-ose/commit/3689df1385b31343ff537334bd3d71cb7f074959

In the implementations I've seen the user all it needs to do is select the client cert throughout the app with file picker and add the cert password

Davx5
![Screenshot_20230404-201308_DAVx⁵.png](https://user-images.githubusercontent.com/52239579/229869224-3c6a4f61-1855-441f-ac33-01cd58dbce7d.png)

Foldersync
![Screenshot_20230404-201409_FolderSync Pro~2.png](https://user-images.githubusercontent.com/52239579/229869298-0eceb1ac-ab67-4c06-bade-dd5173b9c935.png)

Total commander
![Screenshot_20230404-201501_TotalCmd-WebDAV (WEB Folders).png](https://user-images.githubusercontent.com/52239579/229869377-657a1e1f-eff2-4692-996b-b602784f2d51.png)

**Comment by Radiokot** [comment]
Added in [1.3.0](https://github.com/Radiokot/photoprism-android-client/releases/tag/1.3.0(6))

**Reply by ippocratis** [reply]
OMG thanks so mutch!!!
